kindelia_core/src/api.rs

```diff
@@ -289,7 +289,7 @@ pub struct RegInfo {
 pub type ReqAnsSend<T> = oneshot::Sender<T>;
 pub type ReqAnsRecv<T> = oneshot::Receiver<T>;
 
-type PublishResults = Vec<Result<(), ()>>;
+type PublishResults = Vec<Result<(), String>>;
```

Comment:
Let's avoid doing this and start explicitly defining the error types as enums. 
e.g. `EffectfFailure`.

Reply:
Yes, I would suggest that

Reply:
perhaps consider use of thiserror crate.

Reply:
@dan-da `thiserror` seems preety useful tbh. thanks!